### PR TITLE
Post a release note update for release 2.4.0.

### PR DESCRIPTION
**Pull request by kerry-meyer**
##### SUMMARY

Post a release note update for release 2.4.0.

##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".
N/A

| GitHub Issue # |
| -------------- |
| |
N/A

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs

##### OUTPUT

```paste below

```
N/A
##### ADDITIONAL INFORMATION

```paste below

```

##### Checklist:

N/A

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
N/A

### CODE REVIEW

**Review by mingjunzhang2019 — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,33 @@ Dellemc.Enterprise_Sonic Release Notes
...
+
+| Release Date: 2024-0108
+| This release provides an Ansible compliance change required on top of the changes included in
+| the 2.3.0 release of the enterprise_sonic Ansible network resource module collection.
+| It addresses an issue raised by the Ansible core team with the content of the 2.3.0 release,
+| and provides accompanying documentation changes in the README file. Additional details are
+| described below.
+| 1) Update the "requires_ansible" version in the meta/runtime.yml file for this collection
+| to the oldest supported version of ansible-core. (This was recently changed by Redhat/Ansible
+| to version 2.14.0.)
```
Comment (ArunSaravananBalachandran) [inline CHANGELOG.rst:22]:
```suggestion
| to version 2.14.0)
```

File: changelogs/archive_fragments/2.4.0/v2.4.0_summary.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+release_summary: |
+        | Release Date: 2024-0108
+        | This release provides an Ansible compliance change required on top of the changes included in
+        | the 2.3.0 release of the enterprise_sonic Ansible network resource module collection.
+        | It addresses an issue raised by the Ansible core team with the content of the 2.3.0 release,
+        | and provides accompanying documentation changes in the README file. Additional details are
+        | described below.
+        | 1) Update the "requires_ansible" version in the meta/runtime.yml file for this collection
+        | to the oldest supported version of ansible-core. (This was recently changed by Redhat/Ansible
+        | to version 2.14.0.)
```
Comment (ArunSaravananBalachandran) [inline changelogs/archive_fragments/2.4.0/v2.4.0_summary.yaml:10]:
```suggestion
        | to version 2.14.0)
```

File: changelogs/changelog.yaml [hunk 3]
```diff
@@ -673,3 +673,51 @@ releases:
...
+
+        | described below.
+
+        | 1) Update the "requires_ansible" version in the meta/runtime.yml file for
+        this collection
+
+        | to the oldest supported version of ansible-core. (This was recently changed
+        by Redhat/Ansible
+
+        | to version 2.14.0.)
```
Comment (ArunSaravananBalachandran) [inline changelogs/changelog.yaml:703]:
```suggestion
        | to version 2.14.0)
```

**Review by ArunSaravananBalachandran — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst (hunk 1 above)
Comment (kerry-meyer) [reply CHANGELOG.rst:22]: I have enclosed the version number in quotes to separate it from the period at the end of the sentence. I have done this for the Ansible version and Python version mentioned in this release note (for both the changelogs/changelog.yaml file and, via automatic update, the CHANGELOG.rst file).

File: changelogs/archive_fragments/2.4.0/v2.4.0_summary.yaml (hunk 2 above)
Comment (kerry-meyer) [reply changelogs/archive_fragments/2.4.0/v2.4.0_summary.yaml:10]: I have enclosed the version number in quotes to separate it from the period at the end of the sentence. I have done this for the Ansible version and Python version mentioned in this release note (for both the changelogs/changelog.yaml file and, via automatic update, the CHANGELOG.rst file).

File: changelogs/changelog.yaml (hunk 3 above)
Comment (kerry-meyer) [reply changelogs/changelog.yaml:703]: I have enclosed the version number in quotes to separate it from the period at the end of the sentence. I have done this for the Ansible version and Python version mentioned in this release note (for both the changelogs/changelog.yaml file and, via automatic update, the CHANGELOG.rst file).

**Review by ArunSaravananBalachandran — APPROVED** [review]

**Pull request merged**